specklepy/serialization/base_object_serializer.py

```diff
@@ -72,6 +70,10 @@ def traverse_base(self, base: Base) -> Tuple[str, Dict]:
         object_builder.update(speckle_type=base.speckle_type)
         obj, props = base, base.get_serializable_attributes()
 
+        if self.write_transports:
```

Comment:
The objects tests were failing, cause SQLite transport missing objects,
turns out the `write_` and `end_transport` methods were not called in the serializer. I think this was prob a long standing bug.

Reply:
ah! hmm so `begin_write` and `end_write` are called in `operations.send` and `.receive` which is why they're not called in the serializer. however, i think it makes sense to call in the serializer, but it shouldn't be called twice so I will remove them from the `operations`

Reply:
ah wait looking at it again, I don't think it makes sense to add it to `traverse_base` since it gets called recursively so you'll be calling `end_write` multiple times before the write is truly finished

Reply:
If you remove it from here the objects get test fails, so there was some reliance on this happening anyhow. Maybe the test was not testing the right thing 😜

Reply:
which test was it exactly? bc locally i had all tests passing as normal before these changes...